Cython/Compiler/ModuleNode.py

```diff
@@ -2836,13 +3004,22 @@ def generate_pymoduledef_struct(self, env, code):
                        ' as per PEP489"')
         code.putln("#endif")
 
+        code.putln("")
+        code.putln("#if CYTHON_COMPILING_IN_LIMITED_API")
+        code.putln("static void %s_free(void *m) {" % Naming.module_cname)
+        code.putln("  %s_clear(m);" % Naming.module_cname)
+        code.putln("}")
+        code.putln("#endif")
+
         code.putln("")
         code.putln("static struct PyModuleDef %s = {" % Naming.pymoduledef_cname)
         code.putln("  PyModuleDef_HEAD_INIT,")
         code.putln('  %s,' % env.module_name.as_c_string_literal())
         code.putln("  %s, /* m_doc */" % doc)
         code.putln("#if CYTHON_PEP489_MULTI_PHASE_INIT")
         code.putln("  0, /* m_size */")
+        code.putln("#elif CYTHON_COMPILING_IN_LIMITED_API")
+        code.putln("  sizeof(%s), /* m_size */" % env.modulestate_cname)
```

Comment:
All of these changes would eventually also apply to the PEP489 mode, I think. (Just mentioning it here, that doesn't have to be part of this PR.)

Reply:
Very likely! I'll just ignore that for now :)